packages/headless/src/utils/keyboard.ts

```diff
@@ -1,3 +1,26 @@
+import { isSubsetMatching } from "./object";
+
+export type WasKeyPressedOption =
+  | string
+  | Partial<Pick<KeyboardEvent, "altKey" | "key" | "ctrlKey" | "metaKey" | "shiftKey" | "code">>;
+
+/**
+ * Check if a specified key was pressed.
+ * @param event The KeyboardEvent
+ * @param key The key, either the [key property](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key) as a string (e.g. "m")
+ * or an object with the relevant key parameters, e.g. `{ key: "m", altKey: true }`
+ * @returns true, if the key was pressed with the specified parameters
+ */
+export const wasKeyPressed = (event: KeyboardEvent, key: WasKeyPressedOption) => {
```

Comment:
Why is this needed? Couldn't we just check the pressed key?

Reply:
I want to have a convenient way to check for a variety of possible key combinations without a dozen `if (...) else` blocks, e.g. see https://github.com/SchwarzIT/onyx/pull/1160/files#diff-e20cf7a25727ff2875e3ea6f35553d0cef6f794a375ed5b648ce98ba4add672aR24